common/djangoapps/student/tests/test_models.py

```diff
@@ -142,3 +143,16 @@ def test_upgrade_deadline_for_non_upgradeable_enrollment(self, mode):
         """ The property should return None if an upgrade cannot be upgraded. """
         enrollment = CourseEnrollmentFactory(course_id=self.course.id, mode=mode)
         self.assertIsNone(enrollment.upgrade_deadline)
+
+    @skip_unless_lms
+    def test_upgrade_deadline_instructor_paced(self):
+        """ Foo """
+        course = CourseFactory(self_paced=True)
+        CourseModeFactory(
+            course_id=course.id,
+            mode_slug=CourseMode.VERIFIED,
+            # This must be in the future to ensure it is returned by downstream code.
+            expiration_datetime=datetime.datetime.now(pytz.UTC) + datetime.timedelta(days=1)
+        )
+        enrollment = CourseEnrollmentFactory(course_id=course.id, mode=CourseMode.AUDIT)
+        self.assertIsNone(enrollment.upgrade_deadline)
```

Comment:
Shouldn't the enrollment's upgrade_deadline be set, but the schedule's be not set?